### DISCUSSION TITLE
Interesting concern about updating products when in cart

### DISCUSSION THREAD

**Discussion by erickirt** [medusajs/medusa#8569] [Q&A]
If a product is added to the cart, it becomes a line item. What happens if the product sells out by the time of purchase? Does the line item get updated accordingly? Similarly, if the product details change, should the line item reflect these updates as well?

**Comment by ederwii** [comment]
For the second question:
When you have a product already sold in a order, it becomes a lineItem as you mention. This lineItem is a copy of the product/variant that it is at the moment of making the sale. If you modify the product's information, when checking that order details, you can get the snapshot of how the product/variant looked like at that moment, and the id (you can use this id to retrieve the current information if needed).

For the first question:
Are you asking about the process of having a product in the cart, but sold out at the moment of making the purchase? You would expect the order to return an error because there are no more products in the stock? If so, it doesn't happen (in medusa admin site). The API also allows you to make the sell regardless of the available stock quantity. (Actually in the version 1.2x / latest I've reported a bug that stock quantities are not properly set after a sell is made, but an error occur when making a refound/return).

If you are making a storefront on your end, you can make use of the information to display/perform any check in any process, but in the admin that comes with medusa already, it's not built in. Perhaps this will be all resolved in V2.
